rustls/src/crypto/signer.rs

```diff
@@ -111,4 +118,16 @@ impl CertifiedKey {
             .first()
             .ok_or(Error::NoCertificatesPresented)
     }
+
+    /// Verify the consistency of this [`CertifiedKey`]'s public and private keys.
+    /// This is done by performing a comparison of SubjectPublicKeyInfo bytes.
+    pub fn certificate_and_key_are_consistent(&self) -> Result<bool, Error> {
+        Ok(*EndEntityCert::try_from(self.end_entity_cert()?)
+            .map_err(Error::FailedToParseEndEntityCert)?
```

Comment:
Suggest the `ParsedCertificate` wrapper to avoid webpki dependencies leaking out of the `crate::webpki` module.

Reply:
Thanks for the suggestion! I added this in https://github.com/rustls/rustls/pull/1954/commits/38dd19e0ea9471274a9cf67c06b0aad494966a49